src/ImageSharp/Processing/Processors/Convolution/BokehBlurProcessor.cs

```diff
@@ -0,0 +1,534 @@
+using System.Numerics;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+
+using SixLabors.ImageSharp.Memory;
+using SixLabors.ImageSharp.ParallelUtils;
+using SixLabors.ImageSharp.PixelFormats;
+using SixLabors.ImageSharp.Primitives;
+using SixLabors.Memory;
+using SixLabors.Primitives;
+
+namespace SixLabors.ImageSharp.Processing.Processors.Convolution
+{
+    /// <summary>
+    /// Applies bokeh blur processing to the image.
+    /// </summary>
+    /// <typeparam name="TPixel">The pixel format.</typeparam>
+    internal class BokehBlurProcessor<TPixel> : ImageProcessor<TPixel>
+        where TPixel : struct, IPixel<TPixel>
+    {
+        /// <summary>
+        /// The maximum size of the kernel in either direction.
+        /// </summary>
+        private readonly int kernelSize;
+
+        /// <summary>
+        /// The number of components to use when applying the bokeh blur
+        /// </summary>
+        private readonly int componentsCount;
+
+        /// <summary>
+        /// The kernel parameters to use for the current instance (a: X, b: Y, A: Z, B: W)
+        /// </summary>
+        private readonly Vector4[] kernelParameters;
+
+        /// <summary>
+        /// The kernel components for the current instance
+        /// </summary>
+        private readonly Complex64[][] kernels;
+
+        /// <summary>
+        /// The scaling factor for kernel values
+        /// </summary>
+        private readonly float kernelsScale;
+
+        /// <summary>
+        /// The mapping of initialized complex kernels and parameters, to speed up the initialization of new <see cref="BokehBlurProcessor{TPixel}"/> instances
+        /// </summary>
+        private static readonly Dictionary<(int, int), (Vector4[], float, Complex64[][])> Cache = new Dictionary<(int, int), (Vector4[], float, Complex64[][])>();
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="Convolution.BokehBlurProcessor{TPixel}"/> class.
+        /// </summary>
+        /// <param name="radius">
+        /// The 'radius' value representing the size of the area to sample.
+        /// </param>
+        /// <param name="components">
+        /// The number of components to use to approximate the original 2D bokeh blur convolution kernel.
+        /// </param>
+        /// <param name="gamma">
+        /// The gamma highlight factor to use to further process the image.
+        /// </param>
+        public BokehBlurProcessor(int radius = 32, int components = 2, float gamma = 3)
+        {
+            this.Radius = radius;
+            this.kernelSize = (radius * 2) + 1;
+            this.componentsCount = components;
+
+            SixLabors.Guard.MustBeGreaterThanOrEqualTo(gamma, 1, nameof(gamma));
+            this.Gamma = gamma;
+
+            // Reuse the initialized values from the cache, if possible
+            if (Cache.TryGetValue((radius, components), out (Vector4[], float, Complex64[][]) info))
```

Comment:
I'm not sure I understand the issue with the second point. I mean, a `TryGetValue` call is always `O(1)` anyway, and I don't get why would anyone try to exhaust the available RAM by just spamming the bokeh extension with random parameters.
Instead, say someone is really using some parameters that are outside the range we define, wouldn't they always need to recalculate the kernels every time then?
(honest question, I might very well be missing something obvious here)

Reply:
If you expose your parameters through a web API, it might be an attack surface.

Actually, pretty unlikely for a such a fancy processor, so feel free to ignore me, but for more common stuff, like Resize, I would definitely consider this.

Reply:
Mmmh, if you really like the idea of having a limited number of items in the cache, what about capping the cache size at a given size (say, 100, or some other arbitrary value), and using a list that stores the insertion time for each key? This way when the cap is reached, we could throw the oldest key away to make room for the new one. This would tend to keep the most used keys always available.
This simple idea would cost `O(n)` in theory, but if the size is capped that'd technically still be `O(1)`, just with a slightly higher constant.
We could wrap this into a separate type to keep things easy and possibly reusable.
Something like a `PriorityDictionary<TKey,TValue>`, for instance.
Otherwise I can just leave this part as it is in the bokeh processor.
What do you think?

Reply:
Let's keep it simple for now, and worry about this later!

Reply:
Got it, perfect 👍